### PR TITLE
INTLY-7155 Technical documentation of the fields in the Config CRD

### PR DESCRIPTION
**Pull request by Boomatang**
# Description

Feat(Docs): Documentation added for new RHMI config CRD

The documentation that has been added is in the form of comments with in the rhmiconfig_types.go.

**JIRA** https://issues.redhat.com/browse/INTLY-7155

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist
- [ ] This change requires a documentation update 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Verified independently on a cluster by reviewer

### CODE REVIEW

File: pkg/apis/integreatly/v1alpha1/rhmiconfig_types.go [hunk 1]
```diff
@@ -38,20 +38,44 @@ type RHMIConfigStatus struct {
...
-	DuringNextMaintenance bool   `json:"duringNextMaintenance"`
-	ApplyOn               string `json:"applyOn,omitempty"`
+	// contacts: list of contacts which are comma separated
+	// "user1@example.com,user2@example.com"
+	Contacts string `json:"contacts,omitempty"`
+	// always-immediately: boolean value, if set to true an upgrade will be applied as soon as it is available,
+	// whether service affecting or not.
+	// This takes precedences over all other options
+	AlwaysImmediately bool `json:"alwaysImmediately"`
+	// during-next-maintenance: boolean value, if set to an upgrade will be applied with in the next maintenance window.
```
Comment (KevFan) [inline pkg/apis/integreatly/v1alpha1/rhmiconfig_types.go:61]:
```suggestion
	// during-next-maintenance: boolean value, if set to true an upgrade will be applied within the next maintenance window.
```

File: pkg/apis/integreatly/v1alpha1/rhmiconfig_types.go [hunk 2]
```diff
@@ -38,20 +38,44 @@ type RHMIConfigStatus struct {
...
+	// Takes precedence over apply-on
+	DuringNextMaintenance bool `json:"duringNextMaintenance"`
+	// apply-on: string date value. If a 'always-immediately' or 'during-next-maintenance' is not set the customer is
+	// required to pick a time for the upgrade. Time value will be validated byu a webhook and reset to black after
+	// upgrade has completed. Format: "dd MMM YYYY hh:mm" > "12 Jan 1980 23:00". Timezone local to cluster.
```
Comment (KevFan) [inline pkg/apis/integreatly/v1alpha1/rhmiconfig_types.go:66]:
```suggestion
	// apply-on: string date value. If a 'always-immediately' or 'during-next-maintenance' is not set the customer is
	// required to pick a time for the upgrade. Time value will be validated by a webhook and reset to blank after
	// upgrade has completed. Format: "dd MMM YYYY hh:mm" > "12 Jan 1980 23:00". Timezone local to cluster.
```

File: pkg/apis/integreatly/v1alpha1/rhmiconfig_types.go [hunk 3]
```diff
@@ -38,20 +38,44 @@ type RHMIConfigStatus struct {
...
+	ApplyOn string `json:"applyOn,omitempty"`
```
Comment (CathalOConnorRH) [inline pkg/apis/integreatly/v1alpha1/rhmiconfig_types.go:67]: Looks good along with Kevin's changes

```suggestion
	// apply-on: string date value. If 'always-immediately' or 'during-next-maintenance' is not set the customer is
	// required to pick a time for the upgrade. Time value will be validated byu a webhook and reset to black after
	// upgrade has completed. Format: "dd MMM YYYY hh:mm" > "12 Jan 1980 23:00". Timezone local to cluster.
	ApplyOn string `json:"applyOn,omitempty"`
```

**Pull request merged**